### PR TITLE
goodhosts: init at v1.1.1

### PR DESCRIPTION
**Pull request by tchinmai7**
## Description of changes
This PR adds the CLI for [GoodHosts](https://github.com/goodhosts/cli), goodhosts, to nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (cafkafk) [inline maintainers/maintainer-list.nix:15348]: Seems like you missed a semicolon here, maybe this will fix the `Check that maintainer list is sorted` step:
```suggestion
    name = "Tarun Chinmai Sekar";
```

**Review by cafkafk — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix
Comment (schinmai-akamai) [reply maintainers/maintainer-list.nix]: Thanks, fixed now

**Review by cafkafk — CHANGES_REQUESTED** [review]
I see you're still working on getting stuff to work with ofborg, so no need to address this before you've done that.

---

This fails to conform to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

You should change it so that your first commit in this PR is just `maintainers: added <your handle>`, and then all the rest of your changes should be in a single commit `goodhosts-cli: init at 2.12.2`

You'll probably have to remove that merge as well. Good luck!

**Comment by cafkafk** [comment]
If you'd like, you can just define it the old way, and avoid `pkgs/by-name`:



Here's the [link](https://github.com/NixOS/nixpkgs/blob/f4822bb295ea194a1278af287771c615636cbcbf/pkgs/by-name/README.md#manual-migration-guidelines) to that if you wanna read it.

**Comment by schinmai-akamai** [comment]
> If you'd like, you can just define it the old way, and avoid `pkgs/by-name`:
> 
> 
> 
> Here's the [link](https://github.com/NixOS/nixpkgs/blob/f4822bb295ea194a1278af287771c615636cbcbf/pkgs/by-name/README.md#manual-migration-guidelines) to that if you wanna read it.

So I can just throw the package into tools/networking/goodhosts ? 

Thanks for all your help btw - I'm new to nix :)

**Comment by cafkafk** [comment]
> So I can just throw the package into tools/networking/goodhosts ?

Yea, that should be fine!
 
> Thanks for all your help btw - I'm new to nix :)

Nw, I'm glad I can help you :D

File: maintainers/maintainer-list.nix
Comment (cafkafk) [inline maintainers/maintainer-list.nix:15395]: >  error: The option `lib.maintainers.schinmai-akamai.gitubId' does not exist. Definition values:
---
Looks like a spelling mistake, this should fix it!
```suggestion
    githubId = 70169773;
```

Comment (schinmai-akamai) [reply maintainers/maintainer-list.nix]: Great catch!. fixed it

File: pkgs/tools/networking/goodhosts/default.nix
Comment (cafkafk) [inline pkgs/tools/networking/goodhosts/default.nix:24]: You can remove this empty line for consistent formatting.
```suggestion
  '';

  vendorHash = null;
```

File: pkgs/tools/networking/goodhosts/default.nix
Comment (cafkafk) [inline pkgs/tools/networking/goodhosts/default.nix:31]: This can be improved further to conform to the [pkgs/CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes) by adding a `mainProgram` meta attribute.

```suggestion
    maintainers = with maintainers; [ schinmai-akamai ];
    mainProgram = "goodhosts";
  };
```

File: pkgs/tools/networking/goodhosts/default.nix
Comment (cafkafk) [inline pkgs/tools/networking/goodhosts/default.nix:12]: You can use `hash` instead of `sha256`, that is the more idiomatic approach.
```suggestion
    hash = "sha256-BsStTOuMercplx+g+egjiBjCKf5UFCDLGqGVo6i5yKQ=";
```

**Review by cafkafk — APPROVED** [review]
LGTM :+1:. I added some more nitpicks/suggestions that aren't blocking, but you might appreciate knowing for the future.

**Comment by schinmai-akamai** [comment]
@cafkafk thanks for your comments - i've resolved them

File: pkgs/tools/networking/goodhosts/default.nix
Comment (Janik-Haag) [inline pkgs/tools/networking/goodhosts/default.nix:5]: the commit version isn't matching
> [goodhosts-cli: init at 2.12.2](https://github.com/NixOS/nixpkgs/pull/256548/commits/8503166860f7b710508651c5336e92855a07b3c8)

File: pkgs/tools/networking/goodhosts/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/networking/goodhosts/default.nix:6]:
```suggestion
```

File: pkgs/tools/networking/goodhosts/default.nix
Comment (schinmai-akamai) [reply pkgs/tools/networking/goodhosts/default.nix]: Thanks for your comments - fixed up

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 256548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goodhosts</li>
  </ul>
</details>

File: pkgs/tools/networking/goodhosts/default.nix
Comment (Janik-Haag) [inline pkgs/tools/networking/goodhosts/default.nix:1]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
, testers
}:
```
just a small formatting nit

**Review by Janik-Haag — COMMENTED** [review]
Upstream published v1.1.1 a few days ago would you mind updating before we merge this?

File: pkgs/tools/networking/goodhosts/default.nix
Comment (schinmai-akamai) [reply pkgs/tools/networking/goodhosts/default.nix]: done

**Comment by schinmai-akamai** [comment]
> Upstream published v1.1.1 a few days ago would you mind updating before we merge this?

updated it - can you please re-review?

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 256548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>goodhosts</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
there seems to be a hash mismatch

**Comment by schinmai-akamai** [comment]
> there seems to be a hash mismatch
> 
> ```
> error: hash mismatch in fixed-output derivation '/nix/store/5jfxd5dwmbi61jkn3js1h2cv4mwpxiyy-source.drv':
>          specified: sha256-BsStTOuMercplx+g+egjiBjCKf5UFCDLGqGVo6i5yKQ=
>             got:    sha256-VXOMyYM4jS3gYxm3WiKw3uKeC535ppd9iHumPiupGbc=
> error: 1 dependencies of derivation '/nix/store/zlp98m9lzsrgn3swsljc51v3jcbck0b3-goodhosts-1.1.1.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/qyyrbk98p6f5n09anz90rsr8jql5qv9f-review-shell.drv' failed to build
> ```

forgot about it - fixed now

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 256548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>goodhosts</li>
  </ul>
</details>

**Review by cafkafk — COMMENTED** [review]
fails to build for me:
```
❯ nix log /nix/store/y0988yjvgr9hnzcfcrlaicpp4329am29-goodhosts-1.1.1.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/y0988yjvgr9hnzcfcrlaicpp4329am29-goodhosts-1.1.1.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/n5dp7i56fr32frw42kwffpvc6snyhiif-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building subPackage .
go: inconsistent vendoring in /build/source:
        github.com/goodhosts/hostsfile@v0.1.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/olekukonko/tablewriter@v0.0.4: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/sirupsen/logrus@v1.4.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/t-tomalak/logrus-easy-formatter@v0.0.0-20190827215021-c074f06c5816: is explicitly required in go.mod, but not marked as explicit in vend>
        github.com/urfave/cli/v2@v2.1.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/asaskevich/govalidator@v0.0.0-20210307081110-f21760c49a8d: is explicitly required in go.mod, but not marked as explicit in vendor/module>
        github.com/cpuguy83/go-md2man/v2@v2.0.0-20190314233015-f79a8a8ca69d: is explicitly required in go.mod, but not marked as explicit in vendor/modules>
        github.com/dimchansky/utfbom@v1.1.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/konsorten/go-windows-terminal-sequences@v1.0.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/mattn/go-runewidth@v0.0.7: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/russross/blackfriday/v2@v2.0.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/shurcooL/sanitized_anchor_name@v1.0.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/sys@v0.0.0-20190422165155-953cdadca894: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt

        To ignore the vendor directory, use -mod=readonly or -mod=mod.
        To sync the vendor directory, run:
                go mod vendor
```

**Comment by schinmai-akamai** [comment]
This looks like a upstream bug to me, okay if I revert to v1.1.0?

On Oct 7, 2023, at 8:21 PM, Christina Sørensen ***@***.***> wrote:

﻿

@cafkafk commented on this pull request.

fails to build for me:

❯ nix log /nix/store/y0988yjvgr9hnzcfcrlaicpp4329am29-goodhosts-1.1.1.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/y0988yjvgr9hnzcfcrlaicpp4329am29-goodhosts-1.1.1.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/n5dp7i56fr32frw42kwffpvc6snyhiif-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building subPackage .
go: inconsistent vendoring in /build/source:
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vend>
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/module>
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules>
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        ***@***.***: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt

        To ignore the vendor directory, use -mod=readonly or -mod=mod.
        To sync the vendor directory, run:
                go mod vendor

File: pkgs/tools/networking/goodhosts/default.nix
Comment (Janik-Haag) [inline pkgs/tools/networking/goodhosts/default.nix:26]:
```suggestion
  vendorHash = "sha256-pL1z8cfnzcmX5iMVHQJGXYmzGuf8bp4Txbqoh5wSPWQ=";
```
this fixes the build

**Comment by Janik-Haag** [comment]
Also please rename the commit and the pr to `goodhosts: init at v1.1.1`

File: pkgs/tools/networking/goodhosts/default.nix
Comment (schinmai-akamai) [reply pkgs/tools/networking/goodhosts/default.nix]: Thanks!. fixed

**Comment by schinmai-akamai** [comment]
> Also please rename the commit and the pr to `goodhosts: init at v1.1.1`

done - thanks for your review and patience 😄

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 256548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goodhosts</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
> done - thanks for your review and patience 😄
image
I think there went something wrong when you tried renaming the commit it's still the same name ^^

**Pull request merged**
